### ISSUE TITLE
New linter: detect unreachable operands in and and or

### ISSUE THREAD

**Issue by marco-carvalho** [clj-kondo/clj-kondo#2963]
- [ ] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**Is your feature request related to a problem? Please describe.**

A decisive constant before the final operand of `and` or `or` makes all later operands unreachable. This can hide behavior that a developer expected to run.

```clojure
(and false (save! value))
(or :done (save! value))
```

**Describe the solution you'd like**

Add a `:constant-logical-expression` linter, enabled at warning level by default, that reports a decisive non-final constant operand. For `and`, `nil` and `false` are decisive. For `or`, any compile-time truthy value is decisive. Generated forms should not warn.

**Describe alternatives you've considered**

The existing `:constant-condition` linter could be extended, but these are short-circuit expressions rather than predicate positions. A dedicated linter gives users independent control and a specific message.

**Additional context**

Issue #721 concerns constant conditions and unreachable branches, but not short-circuit operands. I am interested in submitting a focused PR with tests and documentation if this approach is accepted.

**Comment by NoahTheDuke** [comment]
I don't think this needs to be a separate linter from `:constant-condition`. When would I want to enable one and disable the other? Why can't the output message from `:constant-condition` differ based on the input type?

**Comment by marco-carvalho** [comment]
I agree. I do not have a compelling configuration case for enabling one and disabling the other. The current PR even has to skip condition position to avoid duplicate reports from `:constant-condition`, which is evidence that the split is artificial. Extending `:constant-condition` with a short-circuit-specific message is the better shape; the separate linter and PR can be replaced by that approach.
